### ISSUE TITLE
Dataviews layouts should have navigable regions

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#70950]
### Description

When the site editor is open in pages or templates view, there are only two or three regions on the screen: "Navigation" (the left sidebar with a black background), "Pages" or "Templates" (the dataview) and, if in List layout, "Editor content". 

The issue is that, within the data view, there are header, main content and footer areas that would benefit from being regions themselves:

<img width="1154" height="787" alt="Image" src="https://github.com/user-attachments/assets/de4c20e1-6ca8-4417-ad8c-70f63c3d1c1a" />

### Step-by-step reproduction instructions

1. Open the site editor and navigate to Templates;
2. Use Ctrl + backtick to navigate between regions and verify there are only two of them.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by youknowriad** [comment]
IMO, regions are App level things and shouldn't be automatically integrated into reusable libraries like DataViews. So I'd consider closing this issue.

If folks really want regions, they can consider using the "compose dataviews UI" approach.

**Comment by tellthemachines** [comment]
Whether it's the actual regions or not we need a way to easily navigate with the keyboard from the middle of a large list of data to the header or the footer. Dataviews implements this header/content/footer layout, so it should be responsible for providing adequate accessibility for it. I'm not super invested in which keyboard navigation mechanism we use here; the regions one would be good for consistency with the rest of the WP UI, but anything is fine as long as it works!

This may not be a huge problem yet, but with an infinitely scrollable list it becomes essential.
